controllers/packetmachine_controller.go

```diff
@@ -203,17 +208,36 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 		}
 	}
 	if dev == nil {
-		// generate a unique UID that will survive pivot, i.e. is not tied to the cluster itself
 		mUID := uuid.New().String()
 		tags := []string{
 			packet.GenerateMachineTag(mUID),
 			packet.GenerateClusterTag(clusterScope.Name()),
 		}
 
-		name := machineScope.Name()
-		dev, err = r.PacketClient.NewDevice(name, clusterScope.PacketCluster.Spec.ProjectID, machineScope, tags)
+		// when the node is a control plan we should check if the elastic ip
+		// for this cluster is not assigned. If it is free we can prepare the
+		// current node to use it.
+		if machineScope.IsControlPlane() {
+			controlPlaneEndpoint, _ = r.PacketClient.GetIPByClusterIdentifier(
+				clusterScope.Namespace(),
+				clusterScope.Name(),
+				clusterScope.PacketCluster.Spec.ProjectID)
+			if len(controlPlaneEndpoint.Assignments) == 0 {
+				a := corev1.NodeAddress{
+					Type:    corev1.NodeExternalIP,
+					Address: controlPlaneEndpoint.Address,
+				}
+				addrs = append(addrs, a)
+				// This tag is used inside the device (via metadata) to configure
+				// the network interface with the elastic ip for this cluster.
+				// In this way the control plane will receive requests made via the control plane endpoint
+				tags = append(tags, fmt.Sprintf("capp.control-plane-endpoint=%s", controlPlaneEndpoint.Address))
```

Comment:
Why is this just on the one node? Shouldn't all of the nodes have this tag?

Reply:
This tag describes who holds the IP. All of the control planes can have the network interface configured but only one has it.

The idea is to use to localize who has the IP. But I am sure this will change when working on #141

Reply:
So this is just informative? OK, good.